snapcraft/plugins/v1/maven.py

```diff
@@ -221,7 +224,16 @@ def build(self):
             arfiles = glob(os.path.join(src, "*.[jw]ar"))
 
             if len(arfiles) == 0:
-                raise RuntimeError("Could not find any built jar files for part")
+                raise RuntimeError(
```

Comment:
Would you mind defining a plugin specific exception in this module and raising that instead. Inherit from SnapcraftException and implement get_brief and get_resolution, something like https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/v1/go.py#L84

Reply:
@sergiusens thanks for the suggestion, that looks way better, I did not knew there was something already made for this purpose of having explanation and solution. I will do it.